src/hotspot/share/gc/g1/g1_globals.hpp

```diff
@@ -158,9 +158,36 @@
           range(0, 100)                                                     \
                                                                             \
   product(uint, G1ExpandByPercentOfAvailable, 20, EXPERIMENTAL,             \
-          "When expanding, % of uncommitted space to claim.")               \
+          "When expanding, % of uncommitted space to expand the heap by in "\
+          "a single expand attempt.")                                       \
           range(0, 100)                                                     \
                                                                             \
+  product(uint, G1ShrinkByPercentOfAvailable, 50, DIAGNOSTIC,               \
+          "When shrinking, maximum % of free space to free for a single "   \
+          "shrink attempt.")                                                \
+          range(0, 100)                                                     \
+                                                                            \
+  product(uint, G1CPUUsageDeviationPercent, 25, DIAGNOSTIC,                 \
+          "The acceptable deviation (in percent) from the target GC CPU "   \
+          "usage (based on GCTimeRatio). Creates a tolerance range "        \
+          "around the target to deal with short-term fluctuations without " \
+          "triggering GC resizing mechanism prematurely.")                  \
+          range(0, 100)                                                     \
+                                                                            \
+  product(uint, G1CPUUsageExpandThreshold, 4, DIAGNOSTIC,                   \
```

Comment:
This, and `G1CPUUsageShrinkThreshold` need to have a range attached to it - if it is set to `0`, I think expansion won't work well initially.

E.g. if set to 0, `_gc_cpu_usage_deviation_counter` will be initialized to 1, and since we compare with `==` in the code

```
  if ((_gc_cpu_usage_deviation_counter == (int)G1CPUUsageExpandThreshold) ||
      (use_long_term_delta && (long_term_gc_cpu_usage > upper_threshold))) {
```

the first term will not fire. Also, we increment `_gc_cpu_usage_deviation_counter` first, then compare.

So a `range(1, MAX_UINT)` should be added to both. Or make the comparison a `>=` or `<=` (for the shrinking).